### DISCUSSION TITLE
Monadic Comprehension Syntax via LINQ in C#

### DISCUSSION THREAD

**Discussion by giscus** [Tyrrrz/Tyrrrzme#29] [Blog]
# Monadic Comprehension Syntax via LINQ in C#

If you ask a C# developer to list the reasons why they enjoy working with the language, they will most likely put LINQ somewhere at the top. LINQ is an extremely convenient set of language tools that provide ways to query and transform data sequences of…

https://tyrrrz.me/blog/monadic-comprehension-via-linq

**Comment by tolgraven** [comment]
Nice!

Thinking about how to put it in Clojure with just straight null, just cause stumbling upon this post made me miss working with it (c# is alright and all but...) and also ponder how exactly it would differ in practice from this whole monad business (which never clicked for me I must admit)

Pretty much to exactly replicate it you'd need some kind of combination version of the some-> and as-> pipes, to bind each step, then just put the some fn last in the pipe, then or none.
Though just some-> and aggregating into a coll as you go then a check that all values are present, and values themselves, then apply... But latter would then get ugly for an example where there's four from captures instead of three, so starting to see the neatness and flexibility of this.
Would be cool if LINQ had more powerful functional batteries included, I'd love to bring something like this into codebase I'm working on but my colleagues would just go wtf, I think. A lot easier to convince them they better get with it if it's built in!

**Comment by tolgraven** [comment]
But at the same time very cool that it can at all be extended to do something like this (and that you did it!) and that the end result code interface is so clean.

heroyam slava

**Comment by ycherkes** [comment]
Oleksii, thank you for the engaging content. Could you relink the "Category Theory via C#" to https://web.archive.org/web/20250609235523/https://weblogs.asp.net/dixin/Tags/Category%20Theory, because the original one has gone away, and ASP.NET Community Blogs now misses the images: https://weblogs.asp.net/dixin

**Reply by Tyrrrz** [reply]
Done, thanks for noticing :)
